### PR TITLE
Allow CLI to generate templates with custom sources

### LINKED ISSUE #2936: [Suggestion] Allows the CLI to generate projects using custom template paths
Projects are now created using the official template by default. People maybe need a basic template without any additional modules, but this one maybe customized. Although it is now possible to do this by modifying the template compression package in the cache, this is extremely inconvenient.

### PR DESCRIPTION
**Pull request by liangshiw**
local source
`abp new Acme.BookStore --template-source  D:\abp\localTemplate`

network source
`abp new Acme.BookStore --template-source https://xxxx.zip`

Resolve #2936

### CODE REVIEW

**Comment by hikalkan** [comment]
Great!

**Comment by maliming** [comment]
@liangshiw  Please update the cli documentation if you like.

**Comment by liangshiw** [comment]
@maliming Ok, I'll update the document

File: framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Commands/Services/SourceCodeDownloadService.cs [hunk 1]
```diff
@@ -38,8 +38,9 @@ public async Task DownloadAsync(string moduleName, string outputFolder, string v
                     version,
                     DatabaseProvider.NotSpecified,
                     UiFramework.NotSpecified,
-                    null,
+                    MobileApp.None,
```
Comment (yekalkan) [inline framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Commands/Services/SourceCodeDownloadService.cs]: Modules don't have MobileApp option. So, setting it null is better.

File: framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectBuilding/TemplateProjectBuilder.cs [hunk 2]
```diff
@@ -82,7 +83,7 @@ public async Task<ProjectBuildResult> BuildAsync(ProjectBuildArgs args)
             var options = args.ExtraProperties
                 .Where(x => !x.Key.Equals(CliConsts.Command, StringComparison.InvariantCultureIgnoreCase))
                 .Where(x => !x.Key.Equals("tiered", StringComparison.InvariantCultureIgnoreCase))
```
Comment (yekalkan) [inline framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectBuilding/TemplateProjectBuilder.cs]: "--template-source" option should be filtered too.

File: framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Commands/NewCommand.cs [hunk 3]
```diff
@@ -271,6 +280,11 @@ public static class Mobile
                 public const string Short = "m";
                 public const string Long = "mobile";
             }
+
+            public static class TemplateSource
+            {
+                public const string Long = "template-source";
```
Comment (yekalkan) [inline framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Commands/NewCommand.cs]: short version may be added: `-ts`

**Comment by liangshiw** [comment]
@yekalkan  All done. 😁

**Pull request merged**